packages/experimental-identity-and-auth/package.json

```diff
@@ -0,0 +1,64 @@
+{
+  "name": "@smithy/experimental-identity-and-auth",
```

Comment:
why does the package name include the word experimental? Doesn't that complicate promoting this to non-experimental later?

Reply:
I was thinking when the feature is ready to move out of experimental (code is stable), this package would be deprecated.

The code from this package would then be moved to the relevant packages (e.g. `@smithy/types`, individual middleware packages, etc.) and be published as production code.

This package should NOT be used in any production code at all.